pkg/build/builder/util.go

```diff
@@ -71,6 +71,10 @@ func getDockerNetworkMode() s2iapi.DockerNetworkMode {
 func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
 	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
 	if err != nil {
+		// for systems without cgroups builds should succeed
+		if _, err := os.Stat("/sys/fs/cgroup"); os.IsNotExist(err) {
+			return &s2iapi.CGroupLimits{}, nil
+		}
```

Comment:
It did for the tests I ran.

On Sat, May 7, 2016 at 7:48 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/build/builder/util.go
> <https://github.com/openshift/origin/pull/8786#discussion_r62424345>:
>
> > @@ -71,6 +71,10 @@ func getDockerNetworkMode() s2iapi.DockerNetworkMode {
> >  func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
> >  	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
> >  	if err != nil {
> > +		// for systems without cgroups builds should succeed
> > +		if _, err := os.Stat("/sys/fs/cgroup"); os.IsNotExist(err) {
> > +			return &s2iapi.CGroupLimits{}, nil
> > +		}
>
> did you check how docker ends up handling these default values? it's going
> to set the memory limit to 0 bytes, and cpu shares/period/quota as well.
> does docker treat 0s for those as unlimited?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8786/files/9725d5171cdd040a8f619a1a7d7b332d9228209b#r62424345>
>